src/zcash/JoinSplit.cpp

```diff
@@ -141,10 +142,26 @@ class JoinSplitCircuit : public JoinSplit<NumInputs, NumOutputs> {
         }
 
         r1cs_ppzksnark_proof<ppzksnark_ppT> r1cs_proof;
-        std::stringstream ss;
-        std::string proof_str(proof.begin(), proof.end());
-        ss.str(proof_str);
-        ss >> r1cs_proof;
+        {
+            LOCK(cs_PointCompression);
```

Comment:
Actually, other users might include the circuit gtest suite, so yeah.

Overall the approach of using globals to change this behavior is a bad one but I just wanted to see how large the transactions were. :)